docs/README.md

```diff
@@ -0,0 +1,624 @@
+        "$BASE_URL/attribute/"
+
+# Inspect all metrics. Notice that metrics have "availability" metadata describing what attributes and normalizations they support.
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/metric/"
+
+# Let's run a query to pull out volume data:
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=date=2018-01"
+# [
+#   {
+#     "volume":12864.75939
+#   }
+# ]
+
+# By default, calculations are "Per Merchant", but another normalization can be selected. Let's try some different metrics with "Per Transaction"
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=date=2018-01&normalization=transaction"
+# [
+#   {
+#     "volume": 73.43840
+#   }
+# ]
+
+# In addition to filtering by a specific month, a range can be provided. Once a date range has been specified, it can be "grouped" so the result set contains the average for each month, instead of the average across the months:
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=2018-01,2018-03&group_by=date"
+# [
+#   {
+#     "date": "2018-01-01",
+#     "volume": 28352.73849
+#   },
+#   {
+#     "date": "2018-02-01",
+#     "volume": 27424.03418
+#   },
+#   {
+#     "date": "2018-03-01",
+#     "volume": 38738.38481
+#   }
+# ]
+
+# Now, let's see the volume for credit and sig debit:
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=date=2018-01;card=credit,sig_debit&group_by=card"
+# [
+#  {
+#    "card": "credit",
+#    "volume": 28327.3061
+#  },
+#  {
+#    "card": "sig_debit",
+#    "volume": 1924.48568
+#  }
+#]
+
+# How about focused on Omaha, Iowa, Kansas, and Missouri?
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=date=2018-01;card=credit,sig_debit;state=MO,KS,NE,IA&group_by=card"
+# [
+#   {
+#     "card": "credit",
+#     "volume": 23394.83581
+#   },
+#   {
+#     "card": "sig_debit",
+#     "volume": 1483.83589
+#   }
+# ]
+
+# Putting these examples together, one can see metrics per transaction in specific states grouped by card and date.
+curl -H "Authorization: Bearer $ID_TOKEN" \
+        "$BASE_URL/query?metrics=volume&filter=date=2018-01,2018-03;card=credit,sig_debit;state=MO,KS,NE,IA&group_by=date,card&normalization=transaction"
+# [
+#   {
+#     "card": "credit",
+#     "date": "2018-01-01",
+#     "volume": 53.77786
+#   },
+#   {
+#     "card": "sig_debit",
+#     "date": "2018-01-01",
+#     "volume": 8.38631
+#   },
+#   {
+#     "card": "credit",
+#     "date": "2018-02-01",
+#     "volume": 68.85441
+#   },
+#   {
+#     "card": "sig_debit",
+#     "date": "2018-02-01",
+#     "volume": 2.41721
+#   },
+#   {
+#     "card": "credit",
+#     "date": "2018-03-01",
+#     "volume": 58.09631
+#   },
+#   {
+#     "card": "sig_debit",
+#     "date": "2018-03-01",
+#     "volume": 5.42187
+#   }
+# ]
+```
+
+### Aggregation
```

Comment:
Everything after this point is more or less "auto" generated from code comments/docs and were originally written mostly for internal use.

@jistas and @alnovacek, as you can see, many of these things are either minimally helpful, confusing, or missing a description entirely. Al has raised a few specific questions, but I think it'd be good if the AIM team as a whole went through each item here (each thing starting with one or more `#`) and made their own definition/description that we should use (maybe even ignoring what's currently written).

There is internal system metadata about each thing that I will add on after the fact (such as if it can be filtered, grouped, possible values, datatype, min/max filter values, etc), but nailing down the high level descriptions will be very important and best left in the AIM team's hands I think.